crates/data-plane-controller/src/controller.rs

```diff
@@ -140,6 +143,12 @@ impl Controller {
         };
         let state = state.as_mut().unwrap();
 
+        let mut none = None;
```

Comment:
This is going to do a full clone of the `ops` repo on every single `on_poll`, which happens _frequently_.

I think this needs a refactor to leverage re-use such as `maybe_checkout`. Can you figure out a nice refactor that has _both_ repos living under sub-directories of `maybe_checkout` ?